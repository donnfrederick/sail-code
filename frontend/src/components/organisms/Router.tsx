import * as Sentry from '@sentry/browser'
import ErrorPage from 'components/pages/Error'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider, Store } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { compose, lifecycle, withState } from 'recompose'
import createStore from 'store/createStore'
import resetViewport from 'utils/resetViewport'
import userAgent from 'utils/userAgent'

declare const gtag: any

export const history = createBrowserHistory()
export const store = createStore(history) as Store<any>

history.listen(() => {
  window.scrollTo(0, 0)
  gtag('config', 'UA-105776977-2', { page_path: history.location.pathname })
  userAgent()
  resetViewport()
})

interface Props {
  children: JSX.Element
  errors: Error | null
  setErrors: (state: any) => any
}

const isProd = process.env.NODE_ENV === 'production'

const sentry = compose(
  withState('errors', 'setErrors', null),
  isProd
    ? lifecycle<Props, {}, {}>({
        componentDidCatch(error, errorInfo) {
          this.props.setErrors(error)
          Sentry.withScope(scope => {
            Object.keys(errorInfo).forEach(key => {
              scope.setExtra(key, errorInfo[key])
            })
            Sentry.captureException(error)
          })
        }
      })
    : lifecycle<Props, {}, {}>({
        componentDidCatch(error) {
          this.props.setErrors(error)
        }
      })
)

export default sentry((props: Props) => {
  const { children, errors } = props

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {errors ? <ErrorPage /> : children}
      </ConnectedRouter>
    </Provider>
  )
})
