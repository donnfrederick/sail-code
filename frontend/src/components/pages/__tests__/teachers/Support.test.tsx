import Router from 'components/organisms/Router'
import TeachersSupport from 'components/pages/teachers/Support'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

test('render <TeachersSupport />', () => {
  const tree = renderer
    .create(
      <Router>
        <TeachersSupport />
      </Router>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
