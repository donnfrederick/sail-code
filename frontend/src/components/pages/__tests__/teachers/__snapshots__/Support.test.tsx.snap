// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <TeachersSupport /> 1`] = `
<div
  className="sc-kkGfuU caxXCF"
>
  <div
    className="sc-hMqMXs hXuWdI"
  >
    <div
      className="sc-kEYyzF eSqWsX"
    >
      <div
        className="sc-cSHVUG jHUXrf"
      >
        <a
          className="sc-kgoBCf jfWKtk"
          href="/teachers/profile"
          onClick={[Function]}
        >
          <div
            className="sc-kAzzGY gxTpSZ"
          >
            自分の情報
          </div>
          <img
            className="sc-chPdSV iwQmvt"
            src="/assets/img/teachers/arrow@3x.png"
          />
        </a>
      </div>
      <div
        className="sc-cSHVUG jHUXrf"
      >
        <a
          className="sc-kgoBCf jfWKtk"
          href="/teachers/history"
          onClick={[Function]}
        >
          <div
            className="sc-kAzzGY gxTpSZ"
          >
            会話履歴
          </div>
          <img
            className="sc-chPdSV iwQmvt"
            src="/assets/img/teachers/arrow@3x.png"
          />
        </a>
      </div>
      <div
        className="sc-cSHVUG jHUXrf"
      >
        <a
          className="sc-kgoBCf jfWKtk"
          href="/teachers/favorites"
          onClick={[Function]}
        >
          <div
            className="sc-kAzzGY gxTpSZ"
          >
            お気に入り一覧
          </div>
          <img
            className="sc-chPdSV iwQmvt"
            src="/assets/img/teachers/arrow@3x.png"
          />
        </a>
      </div>
      <div
        className="sc-cSHVUG jHUXrf"
      >
        <a
          className="sc-kgoBCf jfWKtk"
          href="/teachers/faq"
          onClick={[Function]}
        >
          <div
            className="sc-kAzzGY gxTpSZ"
          >
            よくある質問
          </div>
          <img
            className="sc-chPdSV iwQmvt"
            src="/assets/img/teachers/arrow@3x.png"
          />
        </a>
      </div>
      <div
        className="sc-cSHVUG jHUXrf"
      >
        <a
          className="sc-kgoBCf jfWKtk"
          href="/teachers/terms"
          onClick={[Function]}
        >
          <div
            className="sc-kAzzGY gxTpSZ"
          >
            利用規約
          </div>
          <img
            className="sc-chPdSV iwQmvt"
            src="/assets/img/teachers/arrow@3x.png"
          />
        </a>
      </div>
      <div
        className="sc-cSHVUG jHUXrf"
      >
        <a
          className="sc-kgoBCf jfWKtk"
          href="/teachers/privacy"
          onClick={[Function]}
        >
          <div
            className="sc-kAzzGY gxTpSZ"
          >
            プライバシーポリシー
          </div>
          <img
            className="sc-chPdSV iwQmvt"
            src="/assets/img/teachers/arrow@3x.png"
          />
        </a>
      </div>
      <div
        className="sc-cSHVUG jHUXrf"
      >
        <a
          className="sc-kgoBCf jfWKtk"
          href="/teachers/about"
          onClick={[Function]}
        >
          <div
            className="sc-kAzzGY gxTpSZ"
          >
            運営会社について
          </div>
          <img
            className="sc-chPdSV iwQmvt"
            src="/assets/img/teachers/arrow@3x.png"
          />
        </a>
      </div>
      <div
        className="sc-cSHVUG jHUXrf"
      >
        <a
          className="sc-kgoBCf jfWKtk"
          href="/teachers/blocked"
          onClick={[Function]}
        >
          <div
            className="sc-kAzzGY gxTpSZ"
          >
            ブロックした世界の人
          </div>
          <img
            className="sc-chPdSV iwQmvt"
            src="/assets/img/teachers/arrow@3x.png"
          />
        </a>
      </div>
    </div>
    <div
      className="sc-kGXeez iYgaOI"
      onClick={[Function]}
    >
      ログアウト
    </div>
  </div>
  <button
    className="sc-EHOje fgtSBR"
    data-active={true}
    fontSize={40}
    height={112}
    onClick={[Function]}
    width={384}
  >
    <div
      className="sc-iwsKbI lgKvmV"
    >
      戻る
    </div>
  </button>
  <div
    className="sc-gZMcBi kmcvuH"
  >
    <div
      className="sc-gqjmRU deILtk"
    >
      メニュー
    </div>
    <a
      className="sc-VigVT ewNaTQ"
      href="/teachers/mypage"
      onClick={[Function]}
    >
      ホームに戻る
    </a>
  </div>
</div>
`;
